specification.md

```diff
@@ -0,0 +1,148 @@
+# Specification
+
```

Comment:
We should add either text or a reference to "why this specification exists?", "what's it's intended use?", etc.

Reply:
Agreed. Some sort of preamble is necessary.